explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,405 @@
+target document except for the one that serves the navigation itself.
+    - This is intended to provide a way to transfer the cached signed exchange
+    across origins even if the UA is using origin isolated HTTPCache mechanism.
+    (Note that [header integrity check](#header-integrity-of-signed-exchange)
+    prohibits the distributor from passing a tracking ID to the publisher.)
+- The navigated-to document has a set of preloads for which it uses the
+allowed-alt-sxg link relation to declare that they can be served by signed
+exchanges. The UA either serves all of them from SXGs prefetched by the previous
+page, or none of them. So while processing
+[preload](https://html.spec.whatwg.org/multipage/links.html#link-type-preload)
+link HTTP headers (eg: Link: <https://cdn.publisher.example/lib.js>;
+rel="preload"; as="script"):
+    1. The UA checks whether matching "allowed-alt-sxg" link HTTP header
+    (`Link: <https://cdn.publisher.example/lib.js>;rel="allowed-alt-sxg";header-integrity="sha256-.."`)
+    exists or not. (Note that if the allowed-alt-sxg link HTTP header has
+    variants and variant-key attributes, the UA must execute the algorithm
+    written in
+    [HTTP Representation Variants spec](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+    to find the matching header.)
+    1. The UA checks whether all preload links which have matching
+    allowed-alt-sxg link header have matching (url and header-integrity)
+    parsedExchange which was copied from the referrer page.
+    1. If the check passes, in "linked resource fetch setup steps" for all
+    preload links which have matching allowed-alt-sxg link header the UA sets
+    request's stashed exchange to the matching parsedExchange so the resource
+    will be loaded from the cached signed exchange.
+
+# Detailed design discussion
+## Header integrity of signed exchange
+We propose a new `rel="allowed-alt-sxg"` link header with a `header-integrity`
+parameter. Publishers can declare that the subresource can be served by a signed
+exchange, using this link header in the inner response of the main resource
+signed exchange.
+
+```
+Link: <https://cdn.publisher.example/lib.js>;
+        rel="allowed-alt-sxg";
+        header-integrity="sha256-XXXXXX"
+```
+
+This `header-integrity` parameter is the SHA256 hash value of the
+*signedHeaders* value from the
+[application/signed-exchange format](https://wicg.github.io/webpackage/draft-yasskin-http-origin-signed-responses.html#application-signed-exchange)
+for integrity checking. This signedHeaders is *"the canonical serialization of
+the CBOR representation of the response headers of the exchange represented by
+the application/signed-exchange resource, excluding the Signature header
+field"*. So this value doesn’t change even if the publisher signs the content
+again or changes the signing key. This header-integrity value also guarantees
```

Comment:
```suggestion
again or changes the signing key, but it does change if any of the headers or body change. (It catches changes to the body because a valid signed exchange's headers have to include a `Digest` value that covers the body.)
```